src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== 0) {
+        result += separator;
+      }
+
+      if (this[i] !== null && this[i] !== undefined) {
```

Comment:
Replace `this[i] !== null && this[i] !== undefined` with `this[i]`, it will be the same.

Reply:
> Replace `this[i] !== null && this[i] !== undefined` with `this[i]`, it will be the same.

like this "if (this[i])" its not working